bluetooth: shell: Fix includes

rettichschnidi opened the pull request:
This allows to build the shell with BT_CTLR_DTM and/or BT_CTLR_ADV_EXT enabled.

The issues has been introduced by commit
bf897cf941a514e7ea7a5f837d2287e360281f24 (Bluetooth: Shell: Restructure shell files).

Question:
- For those two settings, would it be okay (tolerated? appreciated?) to add an an extra build in CI "just" to test whether enabling those flags does not break the build?

rettichschnidi committed:
bluetooth: shell: Fix includes

This allows to build the shell with BT_CTLR_DTM and/or BT_CTLR_ADV_EXT
enabled.

The issues has been introduced by commit
bf897cf941a514e7ea7a5f837d2287e360281f24 (Bluetooth: Shell: Restructure
shell files).

Signed-off-by: Reto Schneider <reto.schneider@husqvarnagroup.com>

cvinayak reviewed (changes requested):
> Question:
> 
>     * For those two settings, would it be okay (tolerated? appreciated?) to add an an extra build in CI "just" to test whether enabling those flags does not break the build?

Please add a new `tests` target in `testcases.yaml` file with added `extra_configs` with `CONFIG_BT_CTLR_DTM` for `nrf52840dk/nrf52840` board.

Using which board did you discover the issue?

rettichschnidi commented:
> Using which board did you discover the issue?

An nRF5x based out-of-tree board.

rettichschnidi commented:
> Please add a new `tests` target in `testcases.yaml` file with added `extra_configs` with `CONFIG_BT_CTLR_DTM` for `nrf52840dk/nrf52840` board.

Any proposal on which `testcases.yaml` is should add this to? There are many:
```
$ find tests/bluetooth/controller/ -name testcase.yaml
tests/bluetooth/controller/ctrl_api/testcase.yaml
tests/bluetooth/controller/ctrl_chmu/testcase.yaml
tests/bluetooth/controller/ctrl_cis_create/testcase.yaml
tests/bluetooth/controller/ctrl_cis_terminate/testcase.yaml
tests/bluetooth/controller/ctrl_collision/testcase.yaml
tests/bluetooth/controller/ctrl_conn_update/testcase.yaml
tests/bluetooth/controller/ctrl_cte_req/testcase.yaml
tests/bluetooth/controller/ctrl_data_length_update/testcase.yaml
tests/bluetooth/controller/ctrl_encrypt/testcase.yaml
tests/bluetooth/controller/ctrl_feature_exchange/testcase.yaml
tests/bluetooth/controller/ctrl_hci/testcase.yaml
tests/bluetooth/controller/ctrl_invalid/testcase.yaml
tests/bluetooth/controller/ctrl_le_ping/testcase.yaml
tests/bluetooth/controller/ctrl_min_used_chans/testcase.yaml
tests/bluetooth/controller/ctrl_phy_update/testcase.yaml
tests/bluetooth/controller/ctrl_terminate/testcase.yaml
tests/bluetooth/controller/ctrl_tx_buffer_alloc/testcase.yaml
tests/bluetooth/controller/ctrl_tx_queue/testcase.yaml
tests/bluetooth/controller/ctrl_unsupported/testcase.yaml
tests/bluetooth/controller/ctrl_version/testcase.yaml
tests/bluetooth/controller/ctrl_sca_update/testcase.yaml
tests/bluetooth/controller/ctrl_isoal/testcase.yaml
tests/bluetooth/controller/ctrl_sw_privacy/testcase.yaml
tests/bluetooth/controller/ctrl_sw_privacy_unit/testcase.yaml
tests/bluetooth/controller/ctrl_user_ext/testcase.yaml
tests/bluetooth/controller/ll_settings/testcase.yaml
```

Or did you mean to add a directory `tests/boards/nrf/bluetooth` and put a new `testcase.yaml` in it?

cvinayak commented:
> > Please add a new `tests` target in `testcases.yaml` file with added `extra_configs` with `CONFIG_BT_CTLR_DTM` for `nrf52840dk/nrf52840` board.
> 
> Any proposal on which `testcases.yaml` is should add this to? There are many:
> 
> ```
> $ find tests/bluetooth/controller/ -name testcase.yaml
> tests/bluetooth/controller/ctrl_api/testcase.yaml
> tests/bluetooth/controller/ctrl_chmu/testcase.yaml
> tests/bluetooth/controller/ctrl_cis_create/testcase.yaml
> tests/bluetooth/controller/ctrl_cis_terminate/testcase.yaml
> tests/bluetooth/controller/ctrl_collision/testcase.yaml
> tests/bluetooth/controller/ctrl_conn_update/testcase.yaml
> tests/bluetooth/controller/ctrl_cte_req/testcase.yaml
> tests/bluetooth/controller/ctrl_data_length_update/testcase.yaml
> tests/bluetooth/controller/ctrl_encrypt/testcase.yaml
> tests/bluetooth/controller/ctrl_feature_exchange/testcase.yaml
> tests/bluetooth/controller/ctrl_hci/testcase.yaml
> tests/bluetooth/controller/ctrl_invalid/testcase.yaml
> tests/bluetooth/controller/ctrl_le_ping/testcase.yaml
> tests/bluetooth/controller/ctrl_min_used_chans/testcase.yaml
> tests/bluetooth/controller/ctrl_phy_update/testcase.yaml
> tests/bluetooth/controller/ctrl_terminate/testcase.yaml
> tests/bluetooth/controller/ctrl_tx_buffer_alloc/testcase.yaml
> tests/bluetooth/controller/ctrl_tx_queue/testcase.yaml
> tests/bluetooth/controller/ctrl_unsupported/testcase.yaml
> tests/bluetooth/controller/ctrl_version/testcase.yaml
> tests/bluetooth/controller/ctrl_sca_update/testcase.yaml
> tests/bluetooth/controller/ctrl_isoal/testcase.yaml
> tests/bluetooth/controller/ctrl_sw_privacy/testcase.yaml
> tests/bluetooth/controller/ctrl_sw_privacy_unit/testcase.yaml
> tests/bluetooth/controller/ctrl_user_ext/testcase.yaml
> tests/bluetooth/controller/ll_settings/testcase.yaml
> ```
> 
> Or did you mean to add a directory `tests/boards/nrf/bluetooth` and put a new `testcase.yaml` in it?

This one: https://github.com/zephyrproject-rtos/zephyr/blob/9589e375ab7098a78b72121bb63c8d73d9d54095/tests/bluetooth/shell/testcase.yaml

cvinayak commented:
@rettichschnidi is this something you will be able to follow up soon before the Zephyr v4.0 feature freeze or create GH issue to fix it in bug fix phase?

rettichschnidi commented:
> is this something you will be able to follow up soon

I'll take care of it today.

rettichschnidi committed:
bluetooth: shell: Test BT_CTLR_DTM

This ensures that enabling BT_CTLR_DTM does not break the compilation of
the Bluetooth shell.

Signed-off-by: Reto Schneider <reto.schneider@husqvarnagroup.com>
